scripts/UpdatePackageVersion.swift

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/swift
+
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#if os(Linux)
+import Glibc
+#else
+import Darwin
+#endif
+import class Foundation.PropertyListDecoder
+import class Foundation.ProcessInfo
+import class Foundation.FileManager
+
+// MARK: - Main
+
+/**
+Updates the the version located at the provided file path. 
+The file stored at the path should be a plain text file that only contains the version. 
+The version should be a simple semver formatted version number e.g. MAJOR.MINOR.PATCH
+*/
+func main() {
+  guard CommandLine.arguments.count == 2, let filePath = CommandLine.arguments.last else {
+    printError("A file path must be provided.")
+    return
+  }
+
+  guard let fileContents = FileManager.default.contents(atPath: filePath) else {
+    printError("Failed to retrieve data for file at path \(filePath)")
+    return
+  }
+
+  guard let versionString = String(data: fileContents, encoding: .utf8) else {
+    printError("Failed to retrieve string from file at path: \(filePath)")
+    return
+  }
+
+  guard let version = Version(versionString) else {
+    printError("Failed to parse version from \(versionString)")
+    return
+  }
+
+  var newVersion = version
+  newVersion.incrementMinor()
+
+  do {
+    try newVersion.versionString.write(toFile: filePath , atomically: true, encoding: .utf8)
+  } catch {
+    printError("Failed to write version \(newVersion.versionString) to path \(filePath)")
+  }
+}
+
+// MARK: - Helpers
+
+func printError(_ message: String) {
+  print(message)
+  exit(1)
+}
+
+struct Version {
+  var major: Int
+  var minor: Int
+  var patch: Int
```

Comment:
trivial: private(set)

Reply:
I'd like to allow these properties to be modified if needed. Also will make testing really easy. This is just a dumb simple representation of a Version, no reason to lock them down imo especially since theyre still immutable unless the caller creates a `var`.